src/bindings/swig/ruby/tests/testruby_keyset.rb

```diff
@@ -48,4 +48,102 @@ def test_keySet_append
 
     end  
   end
+
+
+  def test_keySet_each_enumeralbe
+    assert_nothing_raised do
+      a = Array.new
+      # create test keys
+      a << Kdb::Key.new("user/k0", value: "v0", owner: "me")
+      a << Kdb::Key.new("user/k1", value: "v1", owner: "you")
+      a << Kdb::Key.new("user/k2", value: "v2", owner: "john", m2: "a")
+      a << Kdb::Key.new("user/k3", value: "v3", owner: "jane")
+      a << Kdb::Key.new("user/k4", value: "v4", owner: "max")
+      a << Kdb::Key.new("user/k5", value: "v5", owner: "bob", m2: "b")
+      a << Kdb::Key.new("user/k6", value: "v6", owner: "alice")
+      a << Kdb::Key.new("user/k7", value: "v7", owner: "fritz", m3: "c")
+      a << Kdb::Key.new("user/k8", value: "v8", owner: "anton")
+      a << Kdb::Key.new("user/k9", value: "v9", owner: "berta")
+
+      ks = Kdb::KeySet.new
+
+      # populate keySet
+      a.each { |e| ks << e }
```

Comment:
Very elegant! What about testing different ways of keyset construction with always using the checks as below?